src/conventions/h-group/take-action.js

```diff
@@ -230,7 +269,57 @@ export function take_action(game) {
 	let best_play_clue, clue_value;
 	if (state.clue_tokens > 0) {
 		const all_play_clues = play_clues.flat();
-		({ clue: best_play_clue, clue_value } = select_play_clue(all_play_clues));
+		let consider_clues = all_play_clues;
+		const chop = game.me.chop(state.hands[state.ourPlayerIndex]);
+		let saved_clue;
+		let saved_clue_value = -99;
+
+		// Consider saving clues to finesse positions for players who likely have
+		// better cards on chop.
+		if (!state.inEndgame() && !state.early_game && state.clue_tokens < 4 && chop !== undefined) {
+			const our_chop_value = cardValue(state, game.me, chop, chop.order);
+			const better_givers = [];
+			for (let i = 0; i < state.numPlayers; ++i) {
+				if (i == state.ourPlayerIndex)
+					continue;
+				const player = game.players[i];
+				const hand = state.hands[i];
+				if (player.thinksLoaded(state, i, {assume: false}))
+					continue;
+				const otherChop = player.chop(hand);
+				// Saves clue for locked players or players who likely have a better chop.
+				if (otherChop === undefined && player.thinksLocked(state, i))
+					better_givers.push(i);
+				else if (otherChop !== undefined && cardValue(state, player, otherChop, otherChop.order) >= our_chop_value)
+					better_givers.push(i);
+			}
+			if (better_givers.length > 0) {
+				let saved_for = [];
+				consider_clues = consider_clues.filter(clue => {
+					const list = state.hands[clue.target].clueTouched(clue, state.variant).map(c => c.order);
+					const { finesse } = determine_focus(state.hands[clue.target], common, list, {beforeClue: true});
```

Comment:
I'd still prefer if we didn't make the finesse calculation part of `determine_focus()`, since it complicates the function and is only used in this particular case. We could just call `find_finesse()` and compare the orders here?

Reply:
Fair enough, I thought it was somewhat similar to how it returned a bool for whether it was chop focused but understandably that's more important than whether it was finesse focused.